src/Chainweb/Pact/PactService.hs

```diff
@@ -836,8 +837,13 @@ execPreInsertCheckReq txs = pactLabel "execPreInsertCheckReq" $ withDiscardedBat
     withCurrentCheckpointer "execPreInsertCheckReq" $ \pdb -> do
       let v = _chainwebVersion psEnv
           cid = _chainId psEnv
-      liftIO $ fmap Discard $
-        validateChainwebTxs logger v cid cp parentTime currHeight txs (runGas pdb psState psEnv)
+          timeoutLimit = fromIntegral $ (\(Micros n) -> n) $ _psPreInsertCheckTimeout psEnv
+          act = validateChainwebTxs logger v cid cp parentTime currHeight txs (runGas pdb psState psEnv)
+
+      fmap Discard $ liftIO $ timeout timeoutLimit act >>= \case
```

Comment:
>Because System.Timeout.timeout can't interrupt FFI calls.

It can, if they are `interruptible`, but the ones we have in `direct-sqlite` are not.

So yeah, `race` here might be superior in sense that it returns the control to the haskell thread faster (but FFI call might still hang until finished). Replaced, thanks 👍